cast/src/lib.rs

```diff
@@ -121,7 +121,7 @@ pub fn base58_encode(data: &RecordBatch) -> Result<RecordBatch> {
     let mut columns = Vec::<Arc<dyn Array>>::with_capacity(data.columns().len());
 
     for col in data.columns().iter() {
-        if col.data_type() == &DataType::Binary {
```

Comment:
I noticed that after transforming pa.Tables to pl.df in the new join_block_data step, the binary columns gets transformed to large binaries and base58_encoding was not working.

I also need to leave and I pushed to the wrong branch.  Sorry.

Reply:
We also have the same problem for other decode/cast functions. 

Should implement built-in steps in cherry using pyarrow and not polars. And the framework does the conversion properly when using custom steps.

Reply:
I've changed all the functions that compare `col.data_type() == &DataType::Binary`.

> Should implement built-in steps in cherry using pyarrow and not polars.

This follows my comment about more advanced features not supported by PyArrow that we are manually implementing. In this case, it is a simple JOIN using `block_hash` or maybe` block_number`.
I did with one line in polars `joined_df = table_df.join(blocks_df, left_on=config.join_left_on, right_on=config.join_blocks_on, how="left")`. If I were to implement it manually, I would lose all the polar optimizations.